scripts/build_shapes.py

```diff
@@ -105,7 +105,17 @@ def download_GADM(country_code, update=False, out_logging=False):
         #  create data/osm directory
         os.makedirs(os.path.dirname(GADM_inputfile_gpkg), exist_ok=True)
 
-        with requests.get(GADM_url, stream=True) as r:
+        try:
+            r = requests.get(GADM_url, stream=True, timeout=1)
+        except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
+            logger.error(f"GADM server is down at {GADM_url}")
+            raise Exception(f"GADM server is down at {GADM_url}")
```

Comment:
Having logger and raise is a bit annoying.
Could you try if logger.critical(.) or logger.exception(.) also trigger an exit of the execution?
I'd avoid the execution of "raise" because it still leads to dirty logging.
In the general exception tracker, the textual description of the exception `str(exception)` may be added as well.

Reply:
Have played around logger, but there seems to be no way to stop execution just by logging tools :( As far as I was able to find, both `logger.critical( )` and `logger.exception( )` have a purely information function.

Agree  that `raise` doesn't look nice. But executing is anyway stopped when geopandas tries to read a non-existing file. So, probably we can just make an error message more clear. What do you think?

Reply:
An approach may be to use sys.exit() to close the execution forcefully.

Regarding the exception trace, I just realized that there is the option "exc_info=True" that may automatically log the trace.
Ok also as is now, it is is nicer in the log we could also take it into consideration.

What do you think?

Reply:
I think, implementation with `exc_info=True` is definitely nicer :D Revised and ddded `sys.exit(1)`, which seems to give much more clear logging output.